### PR TITLE
Feature/te 471 web fixture writes screenshots

### PR DESCRIPTION
**Pull request by gunther-bachmann**

### CODE REVIEW

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java [hunk 1]
```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
...
+
+	private String runningTest = null;
+	private static final int SCREENSHOT_FILENAME_MAXLEN = 128;
+
+	private String getCurrentTestCase() {
+		return runningTest != null ? runningTest : "UNKNOWN_TEST";
+	}
+
+	private String getScreenshotPath() {
+		// configurable through maven build?
```
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: What should be done for configuring screenshots in Maven, in which file and what? An explanation or a link to a wiki site would be helpful.

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java [hunk 2]
```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
...
+		return "screenshots";
+	}
+
+	private boolean screenshotShouldBeMade(SemanticUnit unit, Action action, String msg) {
+		// configurable through maven build?
```
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: Same here: what should be done for configuring screenshots in Maven, in which file and what? An explanation or a link to a wiki site would be helpful.

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java [hunk 3]
```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
...
+		if (unit == SemanticUnit.TEST && action == Action.ENTER) {
+			runningTest = msg;
+		}
+		if (screenshotShouldBeMade(unit, action, msg)) {
+			screenshot(msg + '.' + action.name());
+		}
+	}
+
+	private String runningTest = null;
+	private static final int SCREENSHOT_FILENAME_MAXLEN = 128;
```
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: Windows 7 has a limitation of a file path length of 252 characters incl. file extension.
So a file name like:
/screenshots/testCaseName_with_128_char_182225.123-escapedBaseName_with128_char.png
could lead to  unpredictable results like, files can not be backed up on other directories, or even worse just could not be opened.

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java [hunk 4]
```diff
@@ -145,10 +154,89 @@ public WebDriver startBrowser(String browser) {
...
+	}
+
+	private String constructScreenshotFilename(String filenameBase, String testcase) {
+		String additionalGraphicType = ".png";
+		String escapedBaseName = filenameBase.replaceAll("[^a-zA-Z0-9.-]", "_").replaceAll("_+", "_")
+				.replaceAll("_+\\.", ".").replaceAll("\\._+", ".");
+		String timeStr = new SimpleDateFormat("HHmmss.SSS").format(new Date());
+		StringBuffer finalFilenameBuffer = new StringBuffer();
+		int lenOfFixedElements = timeStr.length() + additionalGraphicType.length() + 1/* hyphen */;
+		finalFilenameBuffer //
```
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: See above filename limitation under Windows

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java (hunk 1 above)
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: will hopefully be implemented with https://testeditor.atlassian.net/browse/TE-472

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java (hunk 2 above)
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: will hopefully be implemented with https://testeditor.atlassian.net/browse/TE-472

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java (hunk 3 above)
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: The code ensures that each element within a path will be maximal SCREENSHOT_FILENAME_MAXLEN long. This should be ok with Windows 7. SWT-Implementation for screenshots uses the same code which works for windows, apparently.

File: src/main/java/org/testeditor/fixture/web/WebDriverFixture.java (hunk 4 above)
Comment (kbayraktar) [inline src/main/java/org/testeditor/fixture/web/WebDriverFixture.java]: SimpleDateFormat only with Hour, Minute, Seconds and Milliseconds could be more clearly when also a date is specified for detailed description of tests

**Pull request merged**